common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +466,131 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
+
+    DWPAL_SET_SCAN_PARAMS_HELPER_FG(ifname, fg, new_fg);
+    DWPAL_GET_SCAN_PARAMS_HELPER_FG(ifname, out_data, out_len, NL_MAX_REPLY_BUFFSIZE, new_fg);
+
+    DWPAL_SET_SCAN_PARAMS_HELPER_BG(ifname, bg, new_bg);
+    DWPAL_GET_SCAN_PARAMS_HELPER_BG(ifname, out_data, out_len, NL_MAX_REPLY_BUFFSIZE, new_bg);
+
+    if ((new_fg.active != dwell_time_msec) && (new_fg.passive != dwell_time_msec) &&
+        (new_bg.active != dwell_time_msec) && (new_bg.passive != dwell_time_msec)) {
+        LOG(ERROR) << __func__ << " failed to set dwell time =" << dwell_time_msec << std::endl
+                   << "new fg: " << new_fg.passive << " " << new_fg.active << " "
+                   << new_fg.num_probe_reqs << " " << new_fg.probe_reqs_interval << " "
+                   << new_fg.passive_scan_valid_time << " " << new_fg.active_scan_valid_time
+                   << std::endl
+                   << "new bg: " << new_bg.passive << " " << new_bg.active << " "
+                   << new_bg.num_probe_reqs << " " << new_bg.probe_reqs_interval << " "
+                   << new_bg.num_chans_in_chunk << " " << new_bg.break_time << " "
+                   << new_bg.break_time_busy;
+        return false;
+    }
+
+    LOG(DEBUG) << "dwell_time_msec updated from passive = " << fg.passive
+               << " active = " << fg.active << " to = " << dwell_time_msec;
+
+    int freq_index = 0;
+    //configure center frequency for each scanned channel
+    for (auto channel : channel_pool) {
+        //channel validation
+        LOG(DEBUG) << " validating pool channel=" << channel;
+        if (son::wireless_utils::which_freq(m_radio_info.channel) ==
+            beerocks::eFreqType::FREQ_24G) {
+            // 2.4G interface
+            if (son::wireless_utils::which_freq(channel) == beerocks::eFreqType::FREQ_5G) {
+                LOG(ERROR) << " cannot scan 5G channel=" << int(channel)
+                           << " on 2G interface=" << m_radio_info.iface_name;
+                return false;
+            }
+        } else {
+            // 5G interface
+            if (son::wireless_utils::which_freq(channel) == beerocks::eFreqType::FREQ_24G) {
+                LOG(ERROR) << " cannot scan 2G channel=" << int(channel)
+                           << " on 5G interface=" << m_radio_info.iface_name;
+                return false;
+            }
+        }
+
+        dcs_scan_params.freq[freq_index] = beerocks::utils::wifi_channel_to_freq(int(channel));
+        LOG(DEBUG) << " dcs scan pool add center frequency=" << dcs_scan_params.freq[freq_index];
+        freq_index++;
+    }
+
+    dcs_scan_params.ap_force = 1;
+
+    if (dwpal_driver_nl_scan_trigger(get_dwpal_nl_ctx(), (char *)m_radio_info.iface_name.c_str(),
+                                     &dcs_scan_params) != DWPAL_SUCCESS) {
+        LOG(ERROR) << " scan trigger failed! Abort scan ";
+        return false;
+    }
+
+    // timeout for nl confirmation on scan trigger
+    sleep(1);
```

Comment:
Err, that needs more explanation... If we're waiting for confirmation, please use the approprate polling and get calls to actually wait for confirmation.

Reply:
Ported from wav-api.
Nothing to poll on .
We can add comment & comment it out for now, leave the decision for final PR validation stage.

Reply:
> Nothing to poll on .

There are two possibilities.

Either there is a netlink event that confirms the scan was triggered. Then you can do poll()/select() on the fd and get the event.

Or there is no such event but we still need to give the driver some time to register the changed scan parameters. (IOW, the driver is broken.) In that case we need the sleep and a comment that explains why (i.e., the first sentence of this paragraph but then formulated a bit better).

If a sleep is needed, please use C++ `std::this_thread::sleep_for`

Reply:
second option,  and i'll explain why .
In current implementation we are changing scan parameters before triggering the scan , so 
we have to restore them no matter what. 
There could be a lot of failure points until the next select that i the reason why "wait" was chosen.

Reply:
changed the sleep to use the sleep_for format